### PR TITLE
Add end date

### PR DESCRIPTION
**Pull request by dsprayberry**
# Description of change
Adds the use of the optional end-date config property to report sync.

# QA steps
 - [ ] automated tests passing
 - [X] manual qa steps passing (list below)
   - [X] wrote unit tests
   - [X] unit tests are passing
   
# Risks
- minimal; alpha integration

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_sync.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        )
+
+        all_queries_requested = []
+        for request_sent in fake_make_request.call_args_list:
+            # The function signature is gas, query, customer_id
+            _, query, _ = request_sent.args
+            all_queries_requested.append(query)
+
+        delta_days = end_date - start_date
+        expected_days = [start_date]
```
Comment (dmosorast) [inline tests/unittests/test_sync.py:56]: Expectations should be hard coded rather than generated.

File: tests/unittests/test_sync.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+    @patch('tap_google_ads.streams.make_request')
+    def test_no_end_date(self, fake_make_request):
+        start_date = datetime(2022, 1, 1, 0, 0, 0)
+        end_date = None
+        self.do_thing(start_date, end_date, fake_make_request)
+
+    @patch('tap_google_ads.streams.make_request')
+    def test_end_date_one_day_after_start(self, fake_make_request):
```
Comment (dmosorast) [inline tests/unittests/test_sync.py:81]: These tests are pretty hard to follow, can you pull out the assertions and the expectations to the test functions here and rename `do_thing` to `run_sync` or something like that?

**Review by dmosorast — CHANGES_REQUESTED** [review]

**Comment by luandy64** [comment]
We renamed `do_thing` to `run_sync` and made it only set up and run the sync function.

The assertions now live in the test functions.

One of the tests use today as the end date, so that expectation needs to be dynamically generated. But the other test with a defined sync window has hard coded expected query days

**Review by dmosorast — APPROVED** [review]

**Review by dmosorast — APPROVED** [review]
